src/executor/paginator.rs

```diff
@@ -71,7 +71,12 @@ where
             SelectStatement::new()
                 .expr(Expr::cust("COUNT(*) AS num_items"))
                 .from_subquery(
-                    self.query.clone().reset_limit().reset_offset().to_owned(),
+                    self.query
+                        .clone()
+                        .reset_limit()
+                        .reset_offset()
+                        .clear_order_by()
+                        .to_owned(),
```

Comment:
Only these lines are the main change. Others are changes to fix clippy&compile errors.